### PR TITLE
Crew quarters enhanced

### PR DESCRIPTION
**Pull request by ImplicitDeny**
Externalize props spawning, add chairs in the restaurant room with proper sit/stand, fix props positions

### CODE REVIEW

File: addons/gsri_opex/functions/crew/fn_crewAddProps.sqf [hunk 1]
```diff
@@ -0,0 +1,24 @@
+if!(isServer) exitWith {};
+
+params["_room", "_type"];
+
+switch(_type) do {
```
Comment (ArwynFr) [inline addons/gsri_opex/functions/crew/fn_crewAddProps.sqf:5]: Longing for real OOP

File: addons/gsri_opex/functions/crew/fn_crewAddProps.sqf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+		private _mapName = ["Wall_F", format["Wall_%1_F", worldName]] select (toLower worldName in ["altis", "stratis", "malden", "tanoa"]);
+		private _board = (format["Land_MapBoard_01_%1", _mapName]) createVehicle getPos _room;
+		_board attachTo [_room, [-0.721924,-1.7,0.5]];
+		_board setDir 180;
+
+		private _seau = "Land_Bucket_F" createVehicle getPos _room;
+		_seau attachTo [_room, [7.15308,0,-0.7]];
+	};
+	case "B": {
+		private _poubelle = "Land_GarbageBin_01_F" createVehicle getPos _room;
```
Comment (ArwynFr) [inline addons/gsri_opex/functions/crew/fn_crewAddProps.sqf:16]: translate _poubelle to _trashCan ?

File: addons/gsri_opex/functions/fremm/fn_fremmAddCrewQuarters.sqf [hunk 3]
```diff
@@ -9,6 +9,8 @@ if(isServer) then {
 	_roomA setDir (getDir _ship + 90);
 	_roomA call BIS_fnc_Destroyer01PosUpdate;
 	_ship setVariable ["GSRI_FREMM_CrewQuartersRoomA", _roomA, true];
+	[_roomA, "A"] call GSRI_fnc_crewAddProps;
```
Comment (ArwynFr) [inline addons/gsri_opex/functions/fremm/fn_fremmAddCrewQuarters.sqf:12]: In the previous version you created both rooms then added props, now you're changing instructions order.
Did you test it did not cause havoc to add props to room A before creating room B ?

**Review by ArwynFr — DISMISSED** [review]

File: addons/gsri_opex/functions/fremm/fn_fremmAddCrewQuarters.sqf (hunk 3 above)
Comment (ImplicitDeny) [reply addons/gsri_opex/functions/fremm/fn_fremmAddCrewQuarters.sqf:12]: I haven't even thought about it. As far as I can see, this causes absolutely no problem, as long as spawning props for A is done after room A is created. After that all props are attachTo with relative coordinates to their given room, so having room B spawned or not when placing props for A is not relevant.

Seems ok to me \o/

**Review by Seenri — APPROVED** [review]
All good

**Pull request merged**